src/plugins/discover/public/application/main/components/layout/use_discover_histogram.ts

```diff
@@ -126,57 +207,73 @@ export const useDiscoverHistogram = ({
     () =>
       isPlainRecord
         ? undefined
-        : {
-            status: hitsFetchStatus,
+        : localHitsContext ?? {
+            status: hitsFetchStatus.toString() as UnifiedHistogramFetchStatus,
             total: hitsTotal,
           },
-    [hitsFetchStatus, hitsTotal, isPlainRecord]
+    [hitsFetchStatus, hitsTotal, isPlainRecord, localHitsContext]
   );
 
-  const { fetchStatus: chartFetchStatus, response, error } = useDataState(savedSearchData$.charts$);
+  /**
+   * Chart
+   */
 
-  const { bucketInterval, chartData } = useMemo(
-    () =>
-      buildChartData({
-        data,
-        dataView,
-        timeInterval: state.interval,
-        response,
-      }),
-    [data, dataView, response, state.interval]
+  const onChartHiddenChange = useCallback(
+    (chartHidden: boolean) => {
+      // Clear the Lens request adapter when the chart is hidden
+      if (chartHidden) {
+        inspectorAdapters.lensRequests = undefined;
+      }
+
+      storage.set(CHART_HIDDEN_KEY, chartHidden);
+      stateContainer.setAppState({ hideChart: chartHidden });
+    },
+    [inspectorAdapters, stateContainer, storage]
+  );
+
+  const onChartLoad = useCallback(
+    (event: UnifiedHistogramChartLoadEvent) => {
+      // We need to store the Lens request adapter in order to inspect its requests
+      inspectorAdapters.lensRequests = event.adapters.requests;
+    },
+    [inspectorAdapters]
   );
 
+  const [chartHidden, setChartHidden] = useState(state.hideChart);
   const chart = useMemo(
     () =>
       isPlainRecord || !isTimeBased
         ? undefined
         : {
-            status: chartFetchStatus,
-            hidden: state.hideChart,
+            hidden: chartHidden,
             timeInterval: state.interval,
-            bucketInterval,
-            data: chartData,
-            error,
           },
-    [
-      bucketInterval,
-      chartData,
-      chartFetchStatus,
-      error,
-      isPlainRecord,
-      isTimeBased,
-      state.hideChart,
-      state.interval,
-    ]
+    [chartHidden, isPlainRecord, isTimeBased, state.interval]
   );
 
-  return {
-    topPanelHeight,
-    hits,
-    chart,
-    onEditVisualization: canVisualize ? onEditVisualization : undefined,
-    onTopPanelHeightChange,
-    onChartHiddenChange,
-    onTimeIntervalChange,
-  };
+  // state.chartHidden is updated before searchSessionId, which can trigger duplicate
+  // requests, so instead of using state.chartHidden directly, we update chartHidden
+  // when searchSessionId changes
+  useEffect(() => {
+    setChartHidden(state.hideChart);
```

Comment:
This is a particularly icky chunk of code that won't be an issue if we move to manual refetches.